lib/core/src/receive_swap.rs

```diff
@@ -49,46 +62,118 @@ impl ReceiveSwapStateHandler {
+                self.update_swap_info(id, Failed, None).await?;
+                Ok(())
+            }
+
+            // The lockup tx is in the mempool and we accept 0-conf => try to claim
+            // Execute 0-conf preconditions check
+            Ok(RevSwapStates::TransactionMempool) => {
+                let boltzv2::Update::TransactionMempool { transaction, .. } = update else {
+                    return Err(anyhow!("Unexpected payload from Boltz status stream"));
+                };
+
+                let lockup_tx = utils::deserialize_tx_hex(&transaction.hex)?;
+
+                // If the amount is greater than the zero-conf limit
+                // TODO Make limit user-definable through config
+                let receiver_amount_sat = receive_swap.receiver_amount_sat;
+                if receiver_amount_sat > LIQUID_SDK_ZERO_CONF_LIMIT_SAT {
+                    debug!("[Receive Swap {id}] Amount is too high to claim with zero-conf ({receiver_amount_sat} sat > {LIQUID_SDK_ZERO_CONF_LIMIT_SAT} sat). Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Amount is within valid range for zero-conf ({receiver_amount_sat} < {LIQUID_SDK_ZERO_CONF_LIMIT_SAT} sat)");
+
+                // If the transaction has RBF, see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki
+                // TODO: Check for inherent RBF by ensuring all tx ancestors are confirmed
+                let rbf_explicit = lockup_tx.input.iter().any(|input| input.sequence.is_rbf());
+                // let rbf_inherent = lockup_tx_history.height < 0;
+
+                if rbf_explicit {
+                    debug!("[Receive Swap {id}] Lockup transaction signals RBF. Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Lockup tx does not signal RBF. Proceeding...");
+
+                // If the fees are higher than our estimated value
+                let tx_fees: u64 = lockup_tx.all_fees().values().sum();
+                let min_fee_rate = match self.config.network {
+                    Network::Mainnet => LIQUID_SDK_ZERO_CONF_FEE_RATE_MAINNET,
+                    Network::Testnet => LIQUID_SDK_ZERO_CONF_FEE_RATE_TESTNET,
+                };
+                let lower_bound_estimated_fees = lockup_tx.vsize() as f32 * min_fee_rate * 0.8;
+
+                if lower_bound_estimated_fees > tx_fees as f32 {
+                    debug!("[Receive Swap {id}] Lockup tx fees are too low: Expected at least {lower_bound_estimated_fees} sat, got {tx_fees} sat. Waiting for confirmation...");
+                    return Ok(());
+                }
+
+                debug!("[Receive Swap {id}] Lockup tx fees are within acceptable range ({tx_fees} > {lower_bound_estimated_fees} sat). Proceeding with claim.");
+
+                match receive_swap.claim_tx_id {
```

Comment:
Seems validating whether we have or not the claim tx should be at the start of this branch. It is a bit weird to validate the lockup zero conf before we verify that point.